### ISSUE TITLE
Launcher should find Text Editor windows with keyword "text"

### ISSUE THREAD

**Issue by adeverteuil** [pop-os/launcher#167]

**Comment by canadaduane** [comment]
After poking around a bit, I think I understand what's going on here: it turns out the word "text" is a common keyword for many apps, and as a result, there are more than 8 apps being returned [here](https://github.com/pop-os/launcher/blob/master/service/src/lib.rs#L587). Since a maximum of 8 apps are considered for the match, the window is dropped as a candidate, and only the "open (new) app" option remains.

**Comment by SRNissen** [comment]
@canadaduane before I create a new issue then, is my specific problem just more of this same thing?

I have installed KolourPaint, but string `"paint"` doesn't find it:

![image](https://github.com/pop-os/launcher/assets/4851097/1c643e9f-08f0-47fe-92b2-4b3bedd120bc)

Meanwhile the single letter `'k'` finds it  directly.

If so, it might be beneficial for both me and the original issue to change the underlying algorithm (...or introduce yet another customization point) such that it prioritizes applications where the search string is in the name before other applications.

**Comment by canadaduane** [comment]
@SRNissen I don't believe your issue is the same. When I grep for "text" in my installed applications directories, I get several hits:

```
$ grep -inrEl '(keywords|categories|comment|genericname).*\btext' /usr/share/applications/ /usr/local/share/applications/ /usr/share/xsessions/ /var/lib/flatpak/exports/share/applications/ ~/.local/share/applications/ ~/.config/autostart/ ~/.local/share/flatpak/exports/share/applications/

/usr/share/applications/vim.desktop
/usr/share/applications/gnome-universal-access-panel.desktop
/usr/share/applications/codium-url-handler.desktop
/usr/share/applications/libreoffice-writer.desktop
/usr/share/applications/codium.desktop
/usr/share/applications/org.gnome.gedit.desktop
/home/duane/.local/share/applications/blender.desktop
/home/duane/.local/share/applications/codium.desktop
```

Whereas with a search for "paint" in place of "text", I only get 1 result (blender). Unless you have many paint apps installed, it would seem this is caused by something other than the "only consider 8" rule I mentioned above.

**Comment by SRNissen** [comment]
Interesting.

I'm not (Yet! Growth mindset!) skilled enough in Linux dev customs to know if I need to file that as an issue with KolourPaint or with somebody else, do you know?

**Comment by canadaduane** [comment]
> I'm not (Yet! Growth mindset!) skilled enough in Linux dev customs to know if I need to file that as an issue with KolourPaint or with somebody else, do you know?

Did you install it via flatpak (Flathub)? If so, you can visit the page at https://flathub.org/apps/org.kde.kolourpaint. I see "KDE" is the author there, and clicking the link sends me to https://invent.kde.org/teams/flathub. I think you can file an issue there.
